modules/runner/src/main/java/org/apache/ignite/internal/configuration/storage/DistributedConfigurationStorage.java

```diff
@@ -99,6 +106,8 @@ public class DistributedConfigurationStorage implements ConfigurationStorage {
      */
     private final AtomicLong changeId = new AtomicLong(0L);
 
+    private final ExecutorService threadPool = Executors.newCachedThreadPool(new NamedThreadFactory("dst-cfg"));
```

Comment:
I wonder if it's ok to have an unbounded thread pool. Tasks submitted to it should be pretty short-lived, but it still a bit scary to have a thread pool which has not limit on the number of threads. How about introducing 10 for starters?